subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -221,8 +252,10 @@ static int aws_fota_on_publish_evt(struct mqtt_client *const client,
 
 	/* If not processing job */
 	/* Receiving a publish on notify_next_topic could be a job */
-	if (topic_len <= NOTIFY_NEXT_TOPIC_MAX_LEN &&
-	    !strncmp(notify_next_topic, topic, topic_len)) {
+	if (((topic_len <= NOTIFY_NEXT_TOPIC_MAX_LEN) &&
```

Comment:
This is hard to read, could you split it in two variables which so set to these expressions and then OR inside the if check?

Reply:
Sure